lightning/src/ln/wire.rs

```diff
@@ -224,7 +226,7 @@ pub fn write<M: Encode + Writeable, W: Writer>(message: &M, buffer: &mut W) -> R
 /// Defines a type-identified encoding for sending messages over the wire.
 ///
 /// Messages implementing this trait specify a type and must be [`Writeable`] to use with [`write()`].
-pub trait Encode {
+pub trait Encode : alloc::fmt::Debug {
```

Comment:
Hmm, if we (eventually) expose `wire` and allow end-users to write custom message types won't we need it in the interface instead of done at the concrete site in `PeerManager`?

Reply:
Hmm.. possibly. May depend on how it is exposed. Would prefer to cross that bridge when we come to it. I have somewhat of an aversion to using supertraits if they aren't necessary.